mzLib/Proteomics/RetentionTimePrediction/Chronologer/ChronologerRetentionTimeEstimator.cs

```diff
@@ -0,0 +1,141 @@
+    /// "Phosphorylation on Y"
+    /// "Accetylation on K"
+    /// "Succinylation on K"
+    /// "Ubiquitination on K"
+    /// "Methylation on K"
+    /// "Dimethylation on K"
+    /// "Trimethylation on K"
+    /// "Methylation on R"
+    /// "Dimethylation on R"
+    /// </code>
+    /// </summary>
+    /// <param name="baseSequence"></param>
+    /// <param name="fullPeptideSequence"></param>
+    /// <returns></returns>
+    public static double PredictRetentionTime(string baseSequence, string fullPeptideSequence)
+    {
+        torch.Tensor tensor = Tensorize(baseSequence, fullPeptideSequence);
+        var model = new Chronologer();
+        var prediction = model.Predict(tensor);
+        return prediction[0].ToDouble();
+    }
+
+    private static torch.Tensor Tensorize(string baseSequence, string fullPeptideSequence)
+    {
+        var fullSequence = fullPeptideSequence.Split(new[] { '[', ']' })
+            .Where(x => !x.Equals("")).ToArray();
+
+
+        //section to remoce type of mod from the sequence
+        //e.g. Common Fixed:Carbamidomethyl and only stay with the target aa after the mod
+        for (int i = 0; i < fullSequence.Length; i++)
+        {
+            if (fullSequence[i].Contains(" "))
+            {
+                var tempSeq = fullSequence[i].Split(':');
+                fullSequence[i] = tempSeq[1];
+            }
+        }
+
+        if (baseSequence.Length <= 50) //Chronologer only takes sequences of length 50 or less
+        {
+            var tensor = torch.zeros(1, 52, torch.ScalarType.Int64);
+
+            tensor[0][0] = 38; //C-terminus
+            var tensorCounter = 1; //skips the first element which is the C-terminus in the tensor
+            char modID = ' '; //takes the target aa from inside the loop to hold it for the next loop
+
+            bool mod = fullPeptideSequence[0] == '['; // true if the first loop is a mod
+
+            foreach (var subString in fullSequence)
+            {
+                //if mod, enter
+                if (mod)
+                {
+                    var key = (modID, subString[0].ToString());
+
+                    mod = false; //next iteration is not a mod
+                    continue;
+                }
+
+                if (!subString.Contains(" "))
+                {
+                    //without mods
+                    for (int i = 0; i < subString.Length; i++)
+                    {
+                        tensor[0][tensorCounter] = ChronologerDictionary[(subString[i], "")];
+                        tensorCounter = tensorCounter + 1;
+                    }
+
+                    //save target aa for next loop
+                    modID = subString[subString.Length - 1];
+                }
+
+                mod = true;
+            }
+
+            tensor[0][tensorCounter] = 44; //N-terminus
+
+            return tensor;
+        }
+
+        throw new Exception("Sequence is too long for Chronologer. Expected length less or equal to 50");
```

Comment:
Throwing an exception is a really big decision here, we should discuss this with the larger group

Reply:
I think we need to prevent calls for longer sequences and return null, -1, or similar.

Reply:
Should it return null and specify in the method summary that null means not the expected input?